lib/internal/child_process.js

```diff
@@ -94,10 +94,18 @@ const handleConversion = {
       return handle;
     },
 
-    postSend: function(handle, options) {
-      // Close the Socket handle after sending it
-      if (handle && !options.keepOpen)
-        handle.close();
+    postSend: function(handle, options, target) {
+      // Store the handle after successfully sending it, so it can be closed
+      // when the NODE_HANDLE_ACK is received. If the handle could not be sent,
+      // just close it.
+      if (handle && !options.keepOpen) {
+        if (target) {
+          assert(!target._pendingHandle);
+          target._pendingHandle = handle;
```

Comment:
I thought that was not possible because the messages/handles where stored in `_handleQueue` while there were pending ACK's: https://github.com/nodejs/node/blob/master/lib/internal/child_process.js#L566-575. Is this assumption correct?